### ISSUE TITLE
Name with diacritic not searchable in Browse by Author

### ISSUE THREAD

**Issue by alawvt** [VTUL/vtechworks#682]
There are 98 items in the Browsing by Author index for [Täuber, Uwe C.]( https://vtechworks.lib.vt.edu/browse?type=author&value=T%C3%A4uber%2C+Uwe+C.) However a search for this name in the Browse by Author search box yields "Sorry, there are no results for this browse."

The Browsing by Author results for names begining with "T" include
```
Taylors Elementary School [1]
Tazewell High School [1]
Täuber, Uwe C. [98]
Tbaileh, Ahmad Anan [2]
```
So ä comes between a and b.

There are 35 items in the Browsing by Author index for [Tauber, Uwe C.](https://vtechworks.lib.vt.edu/browse?type=author&value=Tauber%2C+Uwe+C.) A search for this name in Browsing by Author search box yields these 35 items.

A search for Täuber in the Advanced search box for author yields 95 results.

My understanding is that both the Advanced Search and the Browse are using the same SOLR search.

**Comment by alawvt** [comment]
This was reported in [DS-4034 The "first few letters" search doesn't work with diacritics](https://jira.lyrasis.org/browse/DS-4034) with reply,

> There are two similar PR to solve the problem related with diacritics:

> https://github.com/DSpace/DSpace/pull/2310

> https://github.com/DSpace/DSpace/pull/2276

> Both should work but I recommend to apply PR 2276 because it should also solve the issue DS-4354

**Comment by alawvt** [comment]
Try adding [ds-4034 #2276](https://github.com/DSpace/DSpace/pull/2276)

(Don't use [ds-4034 #2698](https://github.com/DSpace/DSpace/pull/2698) because this is for DSpace 7.)

**Comment by alawvt** [comment]
[DS-4034 #2276](https://github.com/DSpace/DSpace/pull/2276) affects
```
dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java
dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java
dspace-api/src/main/java/org/dspace/sort/OrderFormatText.java
```

[ DS-4291 Properly show results for 0-9 link in browse #2463 ](https://github.com/DSpace/DSpace/pull/2463) was merged earlier and affects
```
dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java
dspace-api/src/main/java/org/dspace/browse/BrowseInfo.java
dspace-api/src/main/java/org/dspace/browse/SolrBrowseDAO.java
dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/artifactbrowser/ConfigurableBrowse.java
```
Since they both affect `dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java`, I will attempt to patch with both.

**Comment by alawvt** [comment]
This patch strips diacritics in the search term and returns all names with and without the diacritic(s). For example, a search for `Tauber` or a search for `Täuber` both yield the same results shown below.
<img width="1044" alt="DS-4034_PR2276_author_brwose_diacritics" src="https://user-images.githubusercontent.com/13037168/146432033-c439cb9f-71a5-4d13-9bec-9f6cdeadc990.png">
.

**Comment by alawvt** [comment]
When I click on Browse by Author - "0-9", the authors "21st Century Charter School" and "24th Street Baptist Church" are now listed so the 0-9 patch also seems to be working. The LDE is still indexing.

**Comment by alawvt** [comment]
Resolved with PR #779.
